src/main/java/com/mraof/minestuck/computer/DiskBurner.java

```diff
@@ -0,0 +1,50 @@
+package com.mraof.minestuck.computer;
+
+import com.mraof.minestuck.blockentity.ComputerBlockEntity;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.computer.*;
+import com.mraof.minestuck.util.MSTags;
+
+import java.util.ArrayList;
+
+public class DiskBurner extends ButtonListProgram
+{
+	public static final String NAME = "minestuck.disk_burner_program";
+	public static final String BURN_SERVER_DISK = "minestuck.burn_server_disk";
+	public static final String BURN_CLIENT_DISK = "minestuck.burn_client_disk";
+	public static final String CHOOSE = "minestuck.choose";
+	
+	@Override
+	public ArrayList<UnlocalizedString> getStringList(ComputerBlockEntity be)
+	{
+		ArrayList<UnlocalizedString> list = new ArrayList<>();
+		list.add(new UnlocalizedString(CHOOSE));
+		
+		//TODO remove the buttons during an earlier check, it takes two result-less clicks to remove them
```

Comment:
Could you explain what this todo is about?

Reply:
While the code correctly prevents players from burning more disks than there are blank disks stored in the computer, the display of available programs to burn is not immediately updated. What should happen is that as soon as there are no more blank disks available to burn, the buttons for burning disks should dissapear. However it takes multiple clicks for the button display to update

Reply:
Your code advice resolved this issue, thanks!